paragraph/scheduling/instruction_fsm.cc

```diff
@@ -199,4 +202,16 @@ shim::StatusOr<Subroutine*> InstructionFsm::PickSubroutine() {
   return picked_subroutine;
 }
 
+std::string InstructionFsm::ToCsv(const std::string& delimeter) const {
+  CHECK(instruction_->GetGraph() != nullptr);
+  std::string log_entry_str = absl::StrCat(
+      instruction_->GetGraph()->GetProcessorId(), delimeter,
+      instruction_->GetName(), delimeter,
+      OpcodeToString(instruction_->GetOpcode()), delimeter,
+      absl::SixDigits(time_ready_), delimeter,
```

Comment:
I can't find documentation on SixDigits. Will is properly handle the accuracy we need? For example, suppose and instruction started at time 55.0s and ended 10ns later.

Reply:
Well, it looks like the code is the documentation, like [here](https://github.com/abseil/abseil-cpp/blob/master/absl/strings/str_cat.h#L43). Unfortunately it seems that it'll break for less that 6 decimal digits of precision, which is worse than float32 (7.225 digits). I wonder if there is a good library that does double -> string conversion. Or alternatively, protobuf is the answer instead of CSV

Reply:
updated so it gives 12 decimal digits of precision - just enough for 1ps per second